### PR TITLE
Drop obsolete distros from Evergreen config

### PR DESCRIPTION
**Pull request by rcsanchez97**
This PR makes the following changes:

- change macos-1014 EVG distros to macos-1100
- skip the x509 client auth test on macOS (c.f. CDRIVER-4746)
- drop ubuntu1404 variants
- drop debian8 variants

Full Evergreen patch build: https://spruce.mongodb.com/version/652efbe62a60edf10b9ad337/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

### CODE REVIEW

File: .evergreen/legacy_config_generator/evergreen_config_lib/variants.py [hunk 1]
```diff
@@ -53,18 +53,18 @@ def days(n: int) -> int:
             "link-with-cmake-ssl-deprecated",
             "link-with-cmake-snappy",
             "link-with-cmake-snappy-deprecated",
-            OD([("name", "link-with-cmake-mac"), ("distros", ["macos-1014"])]),
-            OD([("name", "link-with-cmake-mac-deprecated"), ("distros", ["macos-1014"])]),
+            OD([("name", "link-with-cmake-mac"), ("distros", ["macos-1100"])]),
```
Comment (kevinAlbs) [inline .evergreen/legacy_config_generator/evergreen_config_lib/variants.py:56]: Update or remove references to `macos-1014` variants found in other files. [Example](https://github.com/mongodb/mongo-c-driver/blob/0e54f99dada9db1196f82a7ef868f58ca8b0f022/.evergreen/config_generator/components/sasl/darwinssl.py#L18-L19).

Update or remove references to `macos-1015` found in other files. CDRIVER-4580 proposes dropping macOS 10.15 too.

Update [platforms.rst](https://github.com/mongodb/mongo-c-driver/blob/master/src/libmongoc/doc/ref/platforms.rst?plain=1#L33) to remove reference to macOS 10.14.

File: .evergreen/legacy_config_generator/evergreen_config_lib/variants.py [hunk 2]
```diff
@@ -80,25 +80,6 @@ def days(n: int) -> int:
         },
         tags=["pr-merge-gate"],
     ),
-    Variant(
-        "clang34ubuntu",
-        "clang 3.4 (Ubuntu 14.04)",
```
Comment (kevinAlbs) [inline .evergreen/legacy_config_generator/evergreen_config_lib/variants.py:85]: Update `platforms.rst` to remove reference to clang 3.4 and clang 3.5.

File: .evergreen/legacy_config_generator/evergreen_config_lib/variants.py [hunk 3]
```diff
@@ -202,18 +182,6 @@ def days(n: int) -> int:
         ],
         {"CC": "gcc"},
     ),
-    Variant(
-        "gcc49",
-        "GCC 4.9 (Debian 8.1)",
```
Comment (kevinAlbs) [inline .evergreen/legacy_config_generator/evergreen_config_lib/variants.py:207]: Update `platforms.rst` to remove reference to GCC 4.9.

**Review by kevinAlbs — CHANGES_REQUESTED** [review]
Suggest also updating the `NEWS` and `src/libbson/NEWS` to note the future announcement of the dropped platforms. Example:


```
libbson 1.25.0 (Unreleased)
===========================

Platform Support:

  * Support for macOS 10.14 is dropped.
  * Support for Ubuntu 14.04 is dropped.
  * Support for Debian 8.1 is dropped.
```

File: .evergreen/legacy_config_generator/evergreen_config_lib/variants.py (hunk 1 above)
Comment (rcsanchez97) [reply .evergreen/legacy_config_generator/evergreen_config_lib/variants.py:56]: Wow, I was so focused on the main Evergreen config that I missed the other occurrences. Change inbound.

**Review by kevinAlbs — APPROVED** [review]

**Pull request merged**